extensions/shared/use-upgrade-flow/index.js

```diff
@@ -38,13 +46,21 @@ export default function useUpgradeFlow( planSlug, onRedirect = noop ) {
 			checkoutUrl: getUpgradeUrl( { plan, planSlug, postId, postType } ),
 			isAutosaveablePost: editorSelector.isEditedPostAutosaveable(),
 			isDirtyPost: editorSelector.isEditedPostDirty(),
+			planData: plan,
 		};
 	}, [] );
 
 	// Alias. Save post by dispatch.
 	const savePost = dispatch( 'core/editor' ).savePost;
 
 	const goToCheckoutPage = async event => {
+		// If checkout overlay is enabled, use it. Otherwise contiue to redirect method.
+		if ( isCheckoutOverlayAvailable() ) {
+			event.preventDefault();
+			window.wp.hooks.doAction( 'a8c.wpcom-block-editor.openCheckoutModal', { products: [planData] } );
+			return;
+		}
```

Comment:
Perhaps if we move this below the `if ( ! window?.top?.location?.href ) {` check we can reuse the `event.preventDefault`.

But this begs the question of what this check `if ( ! window?.top?.location?.href ) {` is used for, probably SSR or E2E? 

In any case, I still think our checkout overlay should appear after this.

Reply:
Looks like `! window?.top?.location?.href` is to check whether or not the block editor is running inside Calypso iframe. See https://github.com/Automattic/jetpack/pull/16797/commits/8f5676046219dc00c5bfb30dd62a1ce9375af886#r469922492
.

(hash anchor doesn't expand collapsed comments, so adding screenshot).
![image](https://user-images.githubusercontent.com/1287077/95459907-e9521f00-097c-11eb-8070-e9f139275cb1.png)

So yes, let's move underneath it.

Reply:
At the moment the conditional checking `window?.top?.location?.href` actually fails (at least whilst sandboxing anyway), and never reaches `event.preventDefault()`. 

This means that the redirect is falling back to the `href={}` on the button instead found [here](https://github.com/Automattic/jetpack/blob/master/extensions/extended-blocks/paid-blocks/upgrade-plan-banner.jsx#L58).

Because of this, and because we don't touch `window` as we're relying on the presence of the hook `a8c.wpcom-block-editor.openCheckoutModal` which indicates whether this checkout overlay is _active_, I've put this additional code above the `window?.top?.location?.href` check.